webapp/channels/src/components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -21,172 +21,150 @@ import {ChannelModerationRoles} from './types';
 
 const PERIOD_TO_SLASH_REGEX = /\./g;
 
-const formattedMessages: any = defineMessages({
+const channelModerationTranslations = ({
     [Permissions.CHANNEL_MODERATED_PERMISSIONS.CREATE_POST]: {
-        title: {
+        title: defineMessage({
             id: t('admin.channel_settings.channel_moderation.createPosts'),
             defaultMessage: 'Create Posts',
-        },
-        description: {
+        }),
+        description: defineMessage({
```

Comment:
I'm not super up to date with the latest plans around i18n and `mmjstool`. I thought I heard we wanted to replace it with built in `react-intl` and am guessing this is part of that effort. If so, do we still need to add support to `mmjstool` in the meantime so it can pick up `defineMessage` strings? e.g. if I do a contrived example like:

```diff
diff --git a/webapp/channels/src/components/app.jsx b/webapp/channels/src/components/app.jsx
index 84f0468bb3..969ec4b0d5 100644
--- a/webapp/channels/src/components/app.jsx
+++ b/webapp/channels/src/components/app.jsx
@@ -13,15 +13,21 @@ import {makeAsyncComponent} from 'components/async_load';

 import CRTPostsChannelResetWatcher from 'components/threading/channel_threads/posts_channel_reset_watcher';
 const LazyRoot = React.lazy(() => import('components/root'));
+import {defineMessage} from 'react-intl';

 const Root = makeAsyncComponent('Root', LazyRoot);

 class App extends React.PureComponent {
     render() {
+        const msg = defineMessage({
+            id: 'test.for.defineMessage',
+            defaultMessage: 'test for defineMessage'
+        });
         return (
             <Provider store={store}>
                 <CRTPostsChannelResetWatcher/>
                 <Router history={getHistory()}>
+                    {msg}
                     <Route
                         path='/'
                         component={Root}
```

The existing `make i18n-extract` job doesn't pick up on the new message.

Reply:
added the support for defineMessages